src/java/htsjdk/samtools/cram/encoding/huffman/codec/HuffmanParamsCalculator.java

```diff
@@ -0,0 +1,153 @@
+package htsjdk.samtools.cram.encoding.huffman.codec;
+
+import htsjdk.samtools.cram.common.MutableInt;
+import htsjdk.samtools.cram.encoding.huffman.HuffmanCode;
+import htsjdk.samtools.cram.encoding.huffman.HuffmanTree;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * The purpose is to calculate Huffman encoding parameters.
+ * Critical: ALL values must be shown to the calculator before calling the calculate() method.
+ */
+public class HuffmanParamsCalculator {
+    private final HashMap<Integer, MutableInt> countMap = new HashMap<Integer, MutableInt>();
```

Comment:
In case of Huffman the map of distinct values should normally be limited to a sensible size, like hundreds of distinct values max. In which case performance of boxed vs primitive int HashMap shouldn't be drastically different.
I think I used to use apache collections here long time ago but given very narrow usage decided to get rid off the dependency. I'd appreciate primitive collections in htsjdk, just not on this PR though :)